main.go

```diff
@@ -241,15 +237,22 @@ func InitFlags(fs *pflag.FlagSet) {
 		"Enable tracing to the opentelemetry-collector service in the same namespace.",
 	)
 
+	AddDiagnosticsOptions(fs, &diagnosticsOptions)
+	logConfig.AddFlags(gofs)
+
 	feature.MutableGates.AddFlag(fs)
 }
 
+// Add RBAC for the authorized diagnostics endpoint.
+// +kubebuilder:rbac:groups=authentication.k8s.io,resources=tokenreviews,verbs=create
+// +kubebuilder:rbac:groups=authorization.k8s.io,resources=subjectaccessreviews,verbs=create
+
 func main() {
-	InitFlags(pflag.CommandLine)
+	InitFlags(pflag.CommandLine, flag.CommandLine)
 	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
 	pflag.Parse()
 
-	ctrl.SetLogger(klogr.New())
+	ctrl.SetLogger(textlogger.NewLogger(logConfig))
```

Comment:
The linter was throwing an error since this pulls in a new version of `klog` which deprecates `klogr.New` so that caused a little bit of extra collateral damage.